doc/webservices/extract_api.rst

```diff
@@ -0,0 +1,486 @@
+                "unit_price": float,
+                "product": string,
+                "taxes_amount": float,
+                "taxes": [
+                            {
+                                "amount": float, 
+                                "type": "fixed"|"percent", 
+                                "price_include": bool
+                            },
+                            ...
+                        ],
+                "subtotal": float,
+                "total": float
+            },
+            ...
+        ]}
+
+Response
+^^^^^^^^
+
+* ``jsonrpc``:  Same as for :ref:`/parse <webservices/extract_api/invoice_parse>`.
+* ``id``:  Same as for :ref:`/parse <webservices/extract_api/invoice_parse>`.
+* ``result``: 
+    * ``status_code``: code indicating the status of the request. ``status_code`` is 0 in case of success. Other ``status_code`` are detailled in the table below.
+    * ``status_msg``: A string giving verbose details about the request status.
+
+=============  ==========================================================
+ status_code       status_msg
+=============  ==========================================================
+0              Success
+12             Validation format is incorrect
+=============  ==========================================================
+
+The ``status_code`` 12 is only used in :ref:`integration mode <webservices/extract_api/integration_testing>`
+
+.. rst-class:: setup doc-aside
+
+.. switcher::
+
+    .. code-block:: json
+
+        {
+            "jsonrpc": string,
+            "id": string,
+            "result": {
+                "status_code": int,
+                "status_msg": string,
+            }
+        }
+
+
+.. _webservices/extract_api/integration_testing:
+
+Integration Testing 
+===================
+
+You can test your integration by using *integration_token* as ``account_token`` in the :ref:`/parse <webservices/extract_api/invoice_parse>` request. This token allows you to simulate the entire flow without really parsing a document.
+
+When calling the :ref:`/parse <webservices/extract_api/invoice_parse>` route with *integration_token* as ``account_token``, the document you send is not parsed. The ``document_id`` you will receive on success is a fake id you can use to test the rest of the flow.
+
+Since the document is not parsed when you call :ref:`/get_results <webservices/extract_api/invoice_get_results>`  you will always receive the same hard coded result. This result follows the response structure described in the documentation.
+
+In integration mode the route :ref:`/validate <webservices/extract_api/invoice_validate>` might return a ``status_code`` 12 to indicate that the format of your validation is incorrect. When using a real ``account_token``, the ``status_code`` will always be 0 even if the format is incorrect.
+
+A python implementation of the full flow can be downloaded :download:`here </webservices/extract_api_implementation.py>`.
```

Comment:
```suggestion
A python implementation of the full flow can be downloaded :download:`here </_static/extract_api_implementation.py>`.
```
Additionally, you must add the following code in `_extensions/odoo_ext/translator.py` (after the `depart_reference` method seems to be a good place) to enable the `download` role:
```py
def visit_download_reference(self, node):
    # type: (nodes.Node) -> None
    if node.hasattr('filename'):
        self.body.append(
            '<a class="reference download internal" href="%s" download="">' %
            posixpath.join(self.builder.dlpath, node['filename']))
        self.body.append(node.astext())
        self.body.append('</a>')
        raise nodes.SkipNode
    else:
        self.context.append('')
def depart_download_reference(self, node):
    # type: (nodes.Node) -> None
    self.body.append(self.context.pop())```
This `translator.py` is our own implementation of the Sphinx translator and it was missing this method, which wasn't the case in the user documentation.

Reply:
Thanks for your help @AntoineVDV